prow/cmd/gcsupload/README.md

```diff
@@ -0,0 +1,38 @@
+# `gcsupload`
+
+`gcsupload` uploads artifacts to cloud storage at a path resolved from the job configuration.
+
+`gcsupload` can be configured by either passing in flags or by specifying a full set of options
+as JSON in the `$GCSUPLOAD_OPTIONS` environment variable, which has the following form:
+
+```json
+{
+    "bucket": "kubernetes-jenkins",
```

Comment:
Can we use gs://kubernetes-jenkins/sub_dir here (and eliminate the subdir key)?

Reply:
No, `subdir` is used primarily when we have multiple containers that are reporting different parts of one test

Reply:
Then can we please name this prefix rather than bucket? We should not restrict users to the root bucket

Reply:
Shouldn't we? The GCS layout right now is pretty stringent that you have `<bucket>/logs`, `<bucket>/pr-logs`...

Reply:
I see no reason this couldn't be:
```
gs://oss-test-results/kubernetes/logs
gs://oss-test-results/kubernetes/pr-logs
gs://oss-test-results/istio/logs
gs://oss-test-results/istio/pr-logs
```

Reply:
This is literally the bucket name, though -- used to get the bucket client from GCS. What is wrong with this approach? If you want an optional prefix before the `logs/` bit that's an RFE?

Reply:
* None of our tooling requires <bucket>/logs vs <prefix>/logs
* This is a regression in how bootstrap works -- https://github.com/kubernetes/test-infra/blob/master/jenkins/bootstrap.py#L625

Reply:
I see, didn't realize it was already in use and supported by bootstrap. Opened https://github.com/kubernetes/test-infra/issues/7566.